esphome/components/voice_assistant/voice_assistant.h

```diff
@@ -90,13 +90,21 @@ class VoiceAssistant : public Component {
 #ifdef USE_SPEAKER
   void set_speaker(speaker::Speaker *speaker) {
     this->speaker_ = speaker;
-    this->local_output_ = true;
+    if (this->speaker_ != nullptr) {
+      this->local_output_ = true;
+    } else {
+      this->local_output_ = false;
+    }
```

Comment:
```suggestion
    this->local_output_ = (this->speaker_ != nullptr)l
```
You can simplify your code into the above.